google/cloud/storage/internal/curl_client.h

```diff
@@ -224,8 +224,13 @@ class CurlClient : public RawClient,
   std::string xml_download_endpoint_;
   std::string iam_endpoint_;
 
+  std::mutex mu_share_;
+  std::mutex mu_dns_;
+  std::mutex mu_ssl_session_;
+  std::mutex mu_connect_;
+  CurlShare share_;
+
   std::mutex mu_;
```

Comment:
Now that this looks like it only protects `generator_`, perhaps a new `GUARDED_BY` comment on that member would be illuminating.

Reply:
Done.